libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -350,7 +350,26 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 						"(remove the \"storage\" keyword)."
 					);
 				if (varLoc == Location::Default || !contract.isLibrary())
+				{
+					if (varLoc == Location::Default)
```

Comment:
Does this have to be inside the other condition in line 352?

Reply:
Yes, because of the or condition, a parameter using memory in a public function would be passed to this if statement, so we need to check if it's using the default location. This is a little confusing, but I didn't want to refactor too much of the existing code.